pkg/telemetry/telemetry.go

```diff
@@ -32,12 +33,15 @@ type engineImpl struct {
 	uuid [16]byte
 	conn pgxconn.PgxConn
 
+	keeper           housekeeper
```

Comment:
Something is strange here... Engine has a reference to housekeeper and housekeeper has a reference to engine. Logically, that doesn't make sense. This needs rethinking. I feel like this is causing a bit of confusion since the responsibilities are not clearly defined.

I can help you there if you feel this comment is a bit overwhelming.

Reply:
I plan to implement all funcs of housekeeper on `engineImpl`. How does that sound?